### PR TITLE
Show email if user don't have name in goal description

### LINKED ISSUE #1710: Show email if user don't have name in goal description
image

Check other places and use helper everywhere

### PR DESCRIPTION
**Pull request by LamaEats**
Fix all places where using user data

## PR includes

- [x] Bug Fix
- [x] Refactor

## Related issues

Resolve #1710 

## QA Instructions, Screenshots, Recordings

image
image
image
image
image
image
image

### CODE REVIEW

File: src/components/UserComboBox.tsx [hunk 1]
```diff
@@ -62,10 +74,20 @@ export const UserComboBox = React.forwardRef<HTMLDivElement, UserComboBoxProps>(
             cb();
         }, []);
 
+        const textValue = useMemo(() => {
+            if (value) {
+                const current = prepareUserDataFromActivity(value);
+
+                return current?.name || current?.email || text;
```
Comment (9teen90nine) [inline src/components/UserComboBox.tsx:81]: `getUserName(current) || text?`

File: src/components/UserGroup.tsx [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+        user?: User | null;
+        ghost?: User | null;
+    }>;
 }
 
 export const UserGroup = ({ users, ...rest }: UserGroupProps) => {
-    const extractedUsers = useMemo(() => users.map((a) => a.user || a.ghost).filter(Boolean) ?? [], [users]);
+    const extractedUsers = useMemo(
+        () =>
+            users.reduce<User>((acc, activity) => {
```
Comment (9teen90nine) [inline src/components/UserGroup.tsx:18]: `users.filter(Boolean).map(prepareUserDataFromActivity)` ?

**Review by 9teen90nine — APPROVED** [review]

File: src/components/UserGroup.tsx (hunk 2 above)
Comment (LamaEats) [reply src/components/UserGroup.tsx:18]: This may be three chain loops insted of one reduce loop

File: src/components/CardHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    timeAgo: React.ReactNode;
+    href?: string;
+}
+
+const StyledCardHeader = styled.div`
+    display: inline-flex;
+`;
+
+const StyledSeparator = styled.span`
+    margin-left: 0.25rem;
```
Comment (awinogradov) [inline src/components/CardHeader.tsx:16]: Use gaps from taskany/colors instead

File: src/components/CardHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+`;
+
+const Separator = () => <StyledSeparator>—</StyledSeparator>;
```
Comment (awinogradov) [inline src/components/CardHeader.tsx:19]: U can use CSS content property and not create this component) but as u wish

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/CardHeader.tsx (hunk 4 above)
Comment (LamaEats) [reply src/components/CardHeader.tsx:19]: No, cause on hover pseudo element will apply text styles from linked element

**Review by awinogradov — APPROVED** [review]

**Pull request merged**